sdks/python/apache_beam/io/gcp/bigquery_tools.py

```diff
@@ -254,11 +258,10 @@ def parse_table_reference(table, dataset=None, project=None):
   # table argument will contain a full table reference instead of just a
   # table name.
   if dataset is None:
-    regex = re.compile(
-        r'''^((?P<project>.+):)?(?P<dataset>\w+)\.
-            (?P<table>[-\w\$]+(\s+\-*\w+)*)$''',
-        re.X)
-    match = regex.match(table)
+    pattern = (
+        f'((?P<project>{_PROJECT_PATTERN})[:\\.])?'
```

Comment:
is a dot an allowed delimiter after project ?

Reply:
is it a common convention elsewhere?

Reply:
dot is actually the only accepted delimiter after project, colon is invalid. https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#table_path

I'm keeping colon as valid here for backwards compatibility.